fitbenchmarking/core/tests/test_fitting_benchmarking_benchmark.py

```diff
@@ -0,0 +1,173 @@
+"""
+Tests for fitbenchmarking.core.fitting_benchmarking.benchmark
+"""
+from __future__ import (absolute_import, division, print_function)
+import inspect
+import copy
+import os
+import unittest
+import mock
+
+from fitbenchmarking import mock_problems
+from fitbenchmarking.utils import fitbm_result
+from fitbenchmarking.core.fitting_benchmarking import benchmark
+from fitbenchmarking.parsing.parser_factory import parse_problem_file
+from fitbenchmarking.utils.options import Options
+from fitbenchmarking.jacobian.SciPyFD_2point_jacobian import ScipyTwoPoint
+from fitbenchmarking.utils.exceptions import NoResultsError
+
+# Defines the module which we mock out certain function calls for
+FITTING_DIR = "fitbenchmarking.core.fitting_benchmarking"
+
+
+def make_fitting_problem(file_name='cubic.dat', minimizers=None):
+    """
+    Helper function that returns a simple fitting problem
+    """
+    options = Options()
+    if minimizers:
+        options.minimizers = minimizers
+
+    bench_prob_dir = os.path.dirname(inspect.getfile(mock_problems))
+    fname = os.path.join(bench_prob_dir, file_name)
+
+    fitting_problem = parse_problem_file(fname, options)
+    fitting_problem.correct_data()
+    jac = ScipyTwoPoint(fitting_problem)
+    fitting_problem.jac = jac
+    return fitting_problem
+
+
+def dict_test(expected, actual):
+    """
+    Test to check two dictionaries are the same
+
+    :param expected: expected dictionary result
+    :type expected: dict
+    :param actual: actual dictionary result
+    :type actual: dict
+    """
+    for key in actual.keys():
+        assert key in expected.keys()
+        assert sorted(actual[key]) == sorted(expected[key])
+
+
+class BenchmarkTests(unittest.TestCase):
+    """
+    benchmark tests
+    """
+
+    def setUp(self):
+        """
+        Setting up problem for tests
+        """
+        self.problem = make_fitting_problem()
+        self.options = Options()
+        self.options.software = ["scipy"]
+        self.scipy_len = len(self.options.minimizers["scipy"])
+        bench_prob_dir = os.path.dirname(inspect.getfile(mock_problems))
+        self.default_parsers_dir = os.path.join(bench_prob_dir,
+                                                "default_parsers")
+        self.all_minimzers = copy.copy(self.options.minimizers)
+
+    def shared_tests(self, expected_names, expected_unselected_minimzers,
+                     expected_minimzers):
+        """
+        Shared tests for the `benchmark` function
```

Comment:
Could a sentence be added to in word say a bit more specifically what shared tests aim to do please

Reply:
Done